source/geometry/FloatBox2D.ooc

```diff
@@ -40,8 +40,8 @@ FloatBox2D: cover {
 	rightCenter ::= FloatPoint2D new(this right, this center y)
 	topCenter ::= FloatPoint2D new(this center x, this top)
 	bottomCenter ::= FloatPoint2D new(this center x, this bottom)
-	empty ::= this size empty
-	area ::= this size x * this size y
+	isEmpty ::= this size hasZeroArea
```

Comment:
It can have zero volume but still contain a singularity and therefore not be empty.